app/models/tslr_claim.rb

```diff
@@ -9,8 +17,30 @@ class TslrClaim < ApplicationRecord
     "2019-2020",
   ].freeze
 
+  enum employment_status: {
```

Comment:
Is there a reason why we're using these Postgres enums as opposed to the standard ActiveRecord enums? The only advantage I can think of is that we're storing a string instead of an integer but when we get the data back from our model ActiveRecord will convert the integer back to a string anyway.

For example, with the code:
```ruby
enum employment_status: [:claim_school, :different_school, :no_school]
```
You could then do
```ruby
claim.employment_status = "claim_school"
```
So `claim.employment_status` would return the string `claim_school` even though it's stored as an integer.

It feels like we're going against the grain slightly with Postgres enums and changing the schema format to SQL could cause issues if we're on slightly different versions of Postgres? For example if I run any migrations a few unrelated lines get removed from the `structure.sql`:

```sql
--
-- Name: plpgsql; Type: EXTENSION; Schema: -; Owner: -
--

CREATE EXTENSION IF NOT EXISTS plpgsql WITH SCHEMA pg_catalog;


--
-- Name: EXTENSION plpgsql; Type: COMMENT; Schema: -; Owner: -
--

COMMENT ON EXTENSION plpgsql IS 'PL/pgSQL procedural language';
```
I'm on Postgres 11.1

Reply:
It's a fair point. I think there are three things we wanted to achieve:

1. use a hash to free the values from the array order
2. use a prefix so we can write `employed_at_claim_school?`, `employed_at_no_school?` and `employed_at_different_school?` which feels more expressive
3. add meaning and security at the db level by storing the values instead of meaningless integers 

I agree the tech overhead seems high, perhaps @tekin can speak to that call?  

This is the prior art we looked at, which is worth a read:

https://naturaily.com/blog/ruby-on-rails-enum

Reply:
I think this is a good point worth discussing.

> It's a fair point. I think there are three things we wanted to achieve:
> 
> 1. use a hash to free the values from the array order
> 2. use a prefix so we can write `employed_at_claim_school?`, `employed_at_no_school?` and `employed_at_different_school?` which feels more expressive

I think this point is moot, we can still use `prefix` and have those expressive boolean methods without using a custom postgres enum type.

> 3. add meaning and security at the db level by storing the values instead of meaningless integers

In my mind, this is the main advantage: that it makes it so the actual value in the database makes more sense. e.g. `"claim_school"` has more meaning and communicates more than `1`. This would be useful say in a situation say where we were exporting all the claims and wanted to do so without the overhead of going through the active record models (@mec as we did [recently on the CCS project](https://github.com/dxw/DataSubmissionServiceAPI/blob/develop/app/models/export/submissions/extract.rb).

> 
> I agree the tech overhead seems high, perhaps @tekin can speak to that call?
> 
> This is the prior art we looked at, which is worth a read:
> 
> https://naturaily.com/blog/ruby-on-rails-enum

As with everything, it's all about tradeoffs! On balance perhaps it's safer to avoid the custom enum type for now. The cost to us is we end up with an ordered array of numbers mapping to a logical value/concept in the database. Perhaps that's OK for now to avoid tying ourselves to a database-specific extension. We can meet halfway by at least being explicit with the integer values we map to:

```ruby
 enum employment_status: {  
  claim_school: 0,
  different_school: 1,
  no_school: 2,
}, _prefix: :employed_at
```

Reply:
I'm happy to drop the DB changes. :+1: